### PR TITLE
[ONNX] Export logical_not

### LINKED ISSUE #95154: RuntimeError: Exporting the operator logical_not to ONNX opset version 9 is not supported.
### 🚀 The feature, motivation and pitch

the logical_or/and/xor are implemented but logical_not is not. Why not cover this operator? Can this be achieved by using other operators?

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by BowenBao**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #96315

Fixes https://github.com/pytorch/pytorch/issues/95154

### CODE REVIEW

File: test/onnx/test_pytorch_onnx_onnxruntime.py [hunk 1]
```diff
@@ -4905,6 +4905,24 @@ def forward(self, x, y):
         y = torch.randint(10, (2, 3, 5), dtype=torch.long)
         self.run_test(XorModel(), input_args=(x, y))
 
+    @skipIfUnsupportedMinOpsetVersion(9)
```
Comment (justinchuby) [inline test/onnx/test_pytorch_onnx_onnxruntime.py:4908]: Enable a opinfo test? I am finishing up https://github.com/pytorch/pytorch/pull/92087 which also fixes the test file.

**Review by justinchuby — APPROVED** [review]

File: test/onnx/test_pytorch_onnx_onnxruntime.py (hunk 1 above)
Comment (BowenBao) [reply test/onnx/test_pytorch_onnx_onnxruntime.py:4908]: Done, very robust this time.

**Pull request closed without merging**